src/framework/pybind_json.hpp

```diff
@@ -347,21 +372,27 @@ void std::from_json(const json_t &js, py::object &o) {
 // Pybind Conversion for Simulator types
 //============================================================================
 
-template<typename T> 
-py::object AerToPy::from_matrix(const matrix<T> &mat) {
-  // THIS SHOULD RETURN A py::array_t but the author was la...
-  size_t rows = mat.GetRows();
-  size_t cols = mat.GetColumns();
-  std::vector<std::vector<T> > tbr;
-  tbr.reserve(rows);
-  for (size_t r = 0; r < rows; r++) {
-    std::vector<T> mrow;
-    mrow.reserve(cols);
-    for (size_t c = 0; c < cols; c++)
-      mrow.emplace_back(mat(r, c));
-    tbr.emplace_back(mrow);
-  }
-  return py::cast(tbr);
+template<typename T>
+py::object AerToPy::array_from_vector(const std::vector<T> &src) {
+    std::array<py::ssize_t, 1> shape { static_cast<py::ssize_t>(src.size()) };
+    auto tbr = py::array_t<T, py::array::f_style>(shape);
+    auto buf = tbr.template mutable_unchecked<1>();
+    for (size_t i = 0; i < src.size(); i++) {
```

Comment:
This definitely wouldn't work - as you mentioned once the vector is destroyed it will free the memory and we'll get a memory fault - btw I started messing around with the vector.data() function here - and I found out recently it does not in fact work (for different reasons) - my first instinct when finding .data() was that it was a bad idea to use it but I ignored that and now I'm rewriting a lot of code... don't repeat my mistake. 
vector is meant to abstract away the memory management when appending/removing elements - it does this by intelligently allocating more memory than necessary (in order to avoid reallocating on every append) but a consequence is that the capacity is often much larger than the number of elements... np.arrays wouldn't like that I don't think.

Reply:
Ok yeah, makes sense.